### PR TITLE
Issuing improved error message when __radd__ called with an integer

### PR DESCRIPTION
**Pull request by jlstevens**
Simple change to address https://github.com/holoviz/holoviews/issues/4834 and help people who try to use `sum` to build `Overlays`.

### CODE REVIEW

**Comment by jlstevens** [comment]
Tests are now passing though I had to include a hackish workaround to bypass an issue [on this line](https://github.com/holoviz/holoviews/blob/master/holoviews/annotators.py#L114). This issue really is a bug in how layouts and overlays are unpacked when composing annotators.

After fixing that and removing the use of `NotImplemented`, this PR should be good to merge.

**Comment by jlstevens** [comment]
Thanks for fixing the failing test! @jbednar Any suggestions for a better error message?

File: holoviews/core/layout.py [hunk 1]
```diff
@@ -23,6 +23,11 @@ def __add__(self, obj):
         "Compose objects into a Layout"
         return Layout([self, obj])
 
+    def __radd__(self, other):
+        if isinstance(other, int):
+            raise TypeError("unsupported operand type(s) for +: 'int' and 'Overlay'. "
+                            "If you are using `sum(elements)` to combine a list of elements,"
+                            "we recommend you to use `Layout(elements)` instead.")
```
Comment (jbednar) [inline holoviews/core/layout.py:30]:
```suggestion
                            "we recommend you to use `Layout(elements)` (and similarly Overlay(elements) for making an overlay from a list) instead.")
```

**Review by jbednar — APPROVED** [review]
Oops! I had filled this out but never submitted my review. Here it is, though the suggested wording probably needs reformatting and to be applied in more places than I indicated.

**Comment by jlstevens** [comment]
Here is an example of the message in a notebook:

image

@philippjfr @jbednar Ready for final review/merge?

**Pull request merged**